src/asim/scripts/resident/2zoneSkim.py

```diff
@@ -112,11 +112,31 @@
 maz_to_maz_walk_cost["DISTWALK"] = net.shortest_path_lengths(maz_to_maz_walk_cost["OMAZ_NODE"], maz_to_maz_walk_cost["DMAZ_NODE"])
 maz_to_maz_walk_cost_out = maz_to_maz_walk_cost[maz_to_maz_walk_cost["DISTWALK"] <= max_maz_maz_walk_dist_feet / 5280.0]
 missing_maz = pd.DataFrame(centroids[~centroids['MAZ'].isin(maz_to_maz_walk_cost_out['OMAZ'])]['MAZ']).rename(columns = {'MAZ': 'OMAZ'}).merge(maz_to_maz_cost[maz_to_maz_cost['OMAZ'] != maz_to_maz_cost['DMAZ']].sort_values('DISTWALK').groupby('OMAZ').agg({'DMAZ': 'first', 'DISTWALK': 'first'}).reset_index(), on = 'OMAZ', how = 'left')
```

Comment:
@aletzdy I am not very clear what is happening here. Please add some comments though you have changed this part in the PR. The line is really long, maybe break it into multiple steps for ease of understanding?

Reply:
Please also make similar update to this line - https://github.com/SANDAG/ABM/blob/ABM3_develop/src/asim/scripts/resident/2zoneSkim.py#L127

Reply:
There is another instance here - https://github.com/SANDAG/ABM/blob/ABM3_develop/src/asim/scripts/resident/2zoneSkim.py#L177. Could we maybe make this a function?

Reply:
Considering that we are using a maximum walk threshold for our walk skims here, it is possible that some MGRAs, especially in non-urban areas, might not be within walking distance of other MGRAs. This means that the skim table may not have walk skims for all MGRAs. What this line is doing is identifying those MGRAs that may be missing from maz_to_maz_walk_cost_out (this is the filteretd by distance skim table), and finding the shortest distance walk distance for them (no to themselves), and therefore creating a missig_maz df that now contains walk skims for those MGRAs that otherwise did not have walk skim at all. I belive ActivitySim required this as of three years ago, but I can check again to see if it is still necessary. Regardless, this line allows for all MGRAs to have a walking skim, even if some are just too far out..

Reply:
I am adding a function to do this job. I suggest we consider refactoring this code for better readbility. It might only take a few hours of work, but will benefit us. I cna create an issue on if you agree.

Reply:
Yes, please create a separate issue to address this. I will merge this PR in about an hour.